modules/misc/xdg-user-dirs.nix

```diff
@@ -0,0 +1,109 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.xdg.userDirs;
+
+in {
+  meta.maintainers = with maintainers; [ pacien ];
+
+  options.xdg.userDirs = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to manage <filename>$XDG_CONFIG_HOME/user-dirs.dirs</filename>.
+        This option requires <option>xdg.enable</option> to be enabled.
+        </para>
+        <para>
+        The generated file is read-only.
```

Comment:
Using `mkOption` instead of `mkEnableOption` here because of the lengthy description.